lib/event-target.js

```diff
@@ -134,44 +137,38 @@ const EventTarget = {
       listener.call(this, new OpenEvent(this));
     }
 
-    const type = options.once ? 'once' : 'on';
-
-    switch (method) {
-      case 'message':
-        onMessage._listener = listener;
-        this[type](method, onMessage);
-        break;
-      case 'close':
-        onClose._listener = listener;
-        this[type](method, onClose);
-        break;
-      case 'error':
-        onError._listener = listener;
-        this[type](method, onError);
-        break;
-      case 'open':
-        onOpen._listener = listener;
-        this[type](method, onOpen);
-        break;
-      default:
-        this[type](method, listener);
-        break;
+    const method = options && options.once ? 'once' : 'on';
+
+    if (type === 'message') {
+      onMessage._listener = listener;
+      this[method](type, onMessage);
+    } else if (type === 'close') {
+      onClose._listener = listener;
+      this[method](type, onClose);
+    } else if (type === 'error') {
+      onError._listener = listener;
+      this[method](type, onError);
+    } else if (type === 'open') {
+      onOpen._listener = listener;
+      this[method](type, onOpen);
+    } else {
+      this[method](type, listener);
```

Comment:
is this faster ? or just less verbose

Reply:
You mean `if...else` instead of `switch`? Last time I checked `switch` was not faster and I find an `if...else` chain easier to read. Furthermore it was already written as an `if...else` chain.